[32mINFO    [0m integration.test_charm:test_charm.py:348 connecting to primary postgresql/5 on 10.241.34.78
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/4 [idle] active: 
  postgresql/5 [idle] active: Primary
  postgresql/6 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/4 [idle] active: 
  postgresql/6 [idle] active: 
  postgresql/7 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/4 [executing] active: 
  postgresql/6 [executing] maintenance: reconfiguring cluster
  postgresql/7 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/7 [executing] waiting: agent initialising
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/4 [executing] active: 
  postgresql/6 [executing] maintenance: reconfiguring cluster
  postgresql/7 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/4 [executing] active: 
  postgresql/6 [executing] maintenance: reconfiguring cluster
  postgresql/7 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/4 [executing] maintenance: reinitialising replica
  postgresql/6 [executing] active: 
  postgresql/7 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/4 [idle] active: 
  postgresql/6 [executing] active: Primary
[32mINFO    [0m integration.test_charm:test_charm.py:368 connecting to the database host: 10.241.34.90
[32mINFO    [0m integration.test_charm:test_charm.py:368 connecting to the database host: 10.241.34.86
[32mINFO    [0m integration.test_charm:test_charm.py:368 connecting to the database host: 10.241.34.124
[32mINFO    [0m pytest_operator.plugin:plugin.py:951 Model status:

Model    Controller     Cloud/Region         Version  SLA          Timestamp
testing  concierge-lxd  localhost/localhost  3.6.8    unsupported  02:03:18Z

App         Version  Status  Scale  Charm       Channel  Rev  Exposed  Message
postgresql  14.18    active      3  postgresql             0  no       

Unit           Workload  Agent  Machine  Public address  Ports     Message
postgresql/4   active    idle   4        10.241.34.90    5432/tcp  
postgresql/6*  active    idle   6        10.241.34.86    5432/tcp  Primary
postgresql/7   active    idle   7        10.241.34.124   5432/tcp  

Machine  State    Address        Inst id        Base          AZ  Message
4        started  10.241.34.90   juju-411ff4-4  ubuntu@22.04      Running
6        started  10.241.34.86   juju-411ff4-6  ubuntu@22.04      Running
7        started  10.241.34.124  juju-411ff4-7  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:957 Juju error logs:

machine-0: 01:32:49 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:32:49 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-0: 01:32:49 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-0: 01:33:29 ERROR unit.postgresql/0.juju-log Unable to get the state of the cluster
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/connection.py", line 198, in _new_conn
    sock = connection.create_connection(
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 85, in create_connection
    raise err
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    response = self._make_request(
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 493, in _make_request
    conn.request(
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/connection.py", line 494, in request
    self.endheaders()
  File "/usr/lib/python3.10/http/client.py", line 1278, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1038, in _send_output
    self.send(msg)
  File "/usr/lib/python3.10/http/client.py", line 976, in send
    self.connect()
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/connection.py", line 325, in connect
    self.sock = self._new_conn()
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/connection.py", line 213, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fb17ef95c30>: Failed to establish a new connection: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 841, in urlopen
    retries = retries.increment(
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/urllib3/util/retry.py", line 519, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='10.241.34.175', port=8008): Max retries exceeded with url: /cluster (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb17ef95c30>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-0/charm/src/cluster.py", line 571, in are_replicas_up
    response = requests.get(
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-0/charm/venv/lib/python3.10/site-packages/requests/adapters.py", line 700, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='10.241.34.175', port=8008): Max retries exceeded with url: /cluster (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb17ef95c30>: Failed to establish a new connection: [Errno 111] Connection refused'))
unit-postgresql-0: 01:33:49 ERROR unit.postgresql/0.juju-log Failed to list PostgreSQL database users: could not translate host name "None" to address: Temporary failure in name resolution

unit-postgresql-0: 01:33:52 ERROR unit.postgresql/0.juju-log upgrade:1: Failed to list PostgreSQL database users: could not translate host name "None" to address: Temporary failure in name resolution

machine-2: 01:34:23 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:34:23 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-2: 01:34:23 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-1: 01:34:29 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:34:29 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-1: 01:34:29 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-1: 01:36:02 ERROR unit.postgresql/1.juju-log Failed to list PostgreSQL database users: connection to server at "10.241.34.196", port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?

unit-postgresql-2: 01:37:05 ERROR unit.postgresql/2.juju-log Failed to list PostgreSQL database users: connection to server at "10.241.34.212", port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?

unit-postgresql-2: 01:38:36 ERROR unit.postgresql/2.juju-log database-peers:2: Failed to list PostgreSQL database users: connection to server at "10.241.34.212", port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?

unit-postgresql-1: 01:42:43 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-2: 01:42:43 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
machine-3: 01:43:28 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:43:28 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-3: 01:43:28 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-4: 01:43:39 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-4: 01:43:39 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-4: 01:43:39 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-4: 01:46:20 ERROR unit.postgresql/4.juju-log Failed to list PostgreSQL database users: connection to server at "10.241.34.90", port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?

unit-postgresql-4: 01:48:56 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-2: 01:48:58 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-3: 01:49:02 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
machine-5: 01:50:22 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-5: 01:50:22 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-5: 01:50:22 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-6: 01:50:39 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-6: 01:50:39 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-6: 01:50:39 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-2: 01:51:36 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-4: 01:51:37 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-4: 01:52:11 ERROR unit.postgresql/4.juju-log database-peers:2: Failed to list PostgreSQL database users: connection to server at "10.241.34.90", port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?

unit-postgresql-6: 01:52:29 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-5: 01:52:29 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-3: 01:52:30 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-4: 01:53:54 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-2: 01:53:57 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
machine-2: 01:55:05 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-5: 01:55:28 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-3: 01:55:29 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-6: 01:55:32 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-6: 01:59:23 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
machine-5: 01:59:35 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
unit-postgresql-6: 01:59:42 ERROR unit.postgresql/6.juju-log Failed to list PostgreSQL database users: could not translate host name "None" to address: Temporary failure in name resolution

unit-postgresql-4: 01:59:45 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent
machine-7: 02:00:51 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-7: 02:00:51 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-7: 02:00:51 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-4: 02:02:12 ERROR unit.postgresql/4.juju-log Failed to list PostgreSQL database users: connection to server at "10.241.34.90", port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?

unit-postgresql-6: 02:02:56 ERROR juju.worker.dependency "log-sender" manifold worker returned unexpected error: sending log message: websocket: close 1005 (no status): websocket: close sent

[32mINFO    [0m pytest_operator.plugin:plugin.py:1039 Forgetting model main...